src/linter.js

```diff
@@ -408,7 +408,14 @@ export default class Linter {
         return this.io.getFiles();
       })
       .then((files) => {
-        return this.scanFiles(Object.keys(files));
+
+        function excludeKnownLibsFilter(file) {
```

Comment:
Avoid inline functions; they will be harder to test but also are harder to follow.

Additionally, I would return a new list of files, then pass that to `scanFiles`. More lines will make things clearer, I think.